src/Author.php

```diff
@@ -33,13 +28,26 @@ class Author extends AbstractEndpoint {
 	 * @return array The array with the data of the endpoint
 	 */
 	public function endpoint_callback( \WP_REST_Request $request ) {
-		$user_id = $request->get_param( 'id' );
-		$response = [];
-		$user = get_user_by( 'id', $user_id );
+		$params = $request->get_params();
+
+		$id = ( false === $params['id'] ) ? false : absint( $params['id'] );
+
+		$slug = ( false === $params['slug'] ) ? false : trim( $params['slug'], '/' );
+
+		if ( false === $id && false === $slug ) {
+			return new \WP_Error( self::INVALID_PARAMS, 'The request must have either an id or a slug', [ 'status' => 400 ] );
```

Comment:
I'm not sure about the other endpoint, but I think if we can simplify something we should do it and open an issue where is not to do it later: 

The example I mentioned before: 

```
$id = ( false === $params['id'] ) ? false : absint( $params['id'] );
$slug = ( false === $params['slug'] ) ? false : trim( $params['slug'], '/' );
 
if ( false === $id && false === $slug ) {
 			return new \WP_Error( self::INVALID_PARAMS, 'The request must have either an id or a slug', [ 'status' => 400 ] );
```

For

```
$id =  absint( $params['id'] );
$slug = trim( sanitize_text_field ( $params['slug'], '/' ) );
 
if ( 0 === $id && '' === $slug ) {
 			return new \WP_Error( self::INVALID_PARAMS, 'The request must have either an id or a slug', [ 'status' => 400 ] );
```

I think we should use the simplest and more elegant solution over any solution that works. As a personal view I see this as a craft not as a job.